tests/Doctrine/Tests/ORM/Functional/Ticket/DDC6499Test.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace Doctrine\Tests\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @group DDC-6499
+ */
+class DDC6499Test extends OrmFunctionalTestCase
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema(
+            [
+            $this->_em->getClassMetadata(DDC6499A::class),
+            $this->_em->getClassMetadata(DDC6499B::class),
+            ]
+        );
+    }
+
+    /**
+     * Test for the bug described in issue #6499.
+     */
+    public function testIssue()
+    {
+        $a = new DDC6499A();
+        $this->_em->persist($a);
+
+        $b = new DDC6499B();
+        $a->setB($b);
+        $this->_em->persist($b);
+
+        $this->_em->flush();
+
+        // Issue #6499 will result in a Integrity constraint violation before reaching this point
+        $this->assertEquals(true, true);
```

Comment:
This should be dropped - instead, try a `find` after a `clear`, and verify that the collection and the association are correctly populated. Put the comment in the assertion message